pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -206,25 +211,36 @@ func (w *Watcher) traversePluginDir(dir string) error {
 			if w.containsBlacklistedDir(path) {
 				return filepath.SkipDir
 			}
-
-			if err := w.fsWatcher.Add(path); err != nil {
-				return fmt.Errorf("failed to watch %s, err: %v", path, err)
-			}
+			dirsToWatch = append(dirsToWatch, path)
 		case mode&os.ModeSocket != 0:
-			w.wg.Add(1)
-			go func() {
```

Comment:
why accumulate rather than continuing to send async here? we can't assume `w.fsWatcher.Events` won't block.

Reply:
Yeah, I noticed that after later...